moto/core/models.py

```diff
@@ -124,31 +124,90 @@ def disable_patching(self):
                      responses.OPTIONS, responses.PATCH, responses.POST, responses.PUT]
 
 
-class ResponsesMockAWS(BaseMockAWS):
+class CallbackResponse(responses.CallbackResponse):
+    '''
+    Need to subclass so we can change a couple things
+    '''
+    def get_response(self, request):
+        '''
+        Need to override this so we can pass decode_content=False
+        '''
+        headers = self.get_headers()
+
+        result = self.callback(request)
+        if isinstance(result, Exception):
+            raise result
+
+        status, r_headers, body = result
+        body = responses._handle_body(body)
+        headers.update(r_headers)
+
+        return responses.HTTPResponse(
+            status=status,
+            reason=six.moves.http_client.responses.get(status),
+            body=body,
+            headers=headers,
+            preload_content=False,
+            # Need to not decode_content to mimic requests
+            decode_content=False,
+        )
+
+    def _url_matches(self, url, other, match_querystring=False):
+        '''
+        Need to override this so we can fix querystrings breaking regex matching
+        '''
+        if not match_querystring:
+            other = other.split('?', 1)[0]
+
+        if responses._is_string(url):
+            if responses._has_unicode(url):
+                url = responses._clean_unicode(url)
+                if not isinstance(other, six.text_type):
+                    other = other.encode('ascii').decode('utf8')
+            return self._url_matches_strict(url, other)
+        elif isinstance(url, responses.Pattern) and url.match(other):
+            return True
+        else:
+            return False
+
 
+botocore_mock = responses.RequestsMock(assert_all_requests_are_fired=False, target='botocore.vendored.requests.adapters.HTTPAdapter.send')
+
+
+class ResponsesMockAWS(BaseMockAWS):
     def reset(self):
+        botocore_mock.reset()
         responses.reset()
```

Comment:
I think we will have problems if multiple clients use `responses`, some of them with a different lifecycle for the responses default mock? I think the `ResponsesMockAWS` should have its own instance of a `RequestsMock`, and use that. The `requests` quickstart teaches people to use the default mock, and if another library like `moto` starts resetting it, then there will be issues.

Reply:
Hmmmm.... I think we will have problems either way, right? `requests` will only be pointed at one or the other. Agree that resetting the default isn't great either though.

Reply:
You're right. I thought `responses` mocks were chained, but unless you explicitly specify a passthrough prefix, they don't. (And I'm not even sure the passthrough will chain properly.)